lib/ssl/src/ssl_handshake.erl

```diff
@@ -1321,13 +1336,28 @@ handle_server_hello_extensions(RecordCB, Random, CipherSuite, Compression,
                                Exts, Version,
 			       #{secure_renegotiate := SecureRenegotation,
                                  next_protocol_selector := NextProtoSelector},
-			       ConnectionStates0, Renegotiation) ->
+			       ConnectionStates0, Renegotiation, IsNew) ->
     ConnectionStates = handle_renegotiation_extension(client, RecordCB, Version,  
                                                       maps:get(renegotiation_info, Exts, undefined), Random, 
 						      CipherSuite, undefined,
 						      Compression, ConnectionStates0,
 						      Renegotiation, SecureRenegotation),
 
+    %% RFC 6066: handle received/expected maximum fragment length
+    if IsNew ->
+            ServerMaxFragEnum = maps:get(max_frag_enum, Exts, undefined),
+            #{current_write := #{max_fragment_length := ConnMaxFragLen}} = ConnectionStates,
+            ClientMaxFragEnum = max_frag_enum(ConnMaxFragLen),
+
+            if ServerMaxFragEnum == ClientMaxFragEnum ->
+                    ok;
+               true ->
+                    throw(?ALERT_REC(?FATAL, ?ILLEGAL_PARAMETER))
+            end;
+       true ->
+            ok
```

Comment:
Maybe we should add a comment here that the `max_fragment_length `extension is ignored if this is a session resumption.

>  The negotiated length applies for the duration of the session
   including session resumptions.